client/my-sites/media-library/content.jsx

```diff
@@ -306,7 +366,10 @@ class MediaLibraryContent extends React.Component {
 	}
 
 	renderHeader() {
-		if ( ! this.props.isConnected && this.needsToBeConnected() ) {
+		if (
+			( ! this.props.isConnected && this.needsToBeConnected() ) ||
+			this.hasGoogleExpired( this.props )
```

Comment:
I'm a bit confused by this check and the `needsToBeConnected` logic. How about making this check

```js
if ( this.needsToBeConnected() ) {
    return null;
}
```

and update the `needsToBeConnected` code to:

```js
needsToBeConnected() {
    const { source, isConnected } = this.props;

    if ( source !== '' && ! isConnected ) {
        return true;
    }

    if ( source !== '' && isConnected && hasGoogleExpired() ) {
        return true;
    }

    return false;
}
```

Reply:
Yep that's a lot clearer - updated!